app/controllers/actions/DataRequiredAction.scala

```diff
@@ -33,7 +33,7 @@ class DataRequiredActionImpl @Inject()(implicit val executionContext: ExecutionC
 
     request.userAnswers match {
       case None =>
-        Future.successful(Left(Redirect(routes.SessionExpiredController.onPageLoad())))
+        Future.successful(Left(throw new Exception("Unable to redirect to page")))
```

Comment:
Hmm - might need to think a little more about this one. This controller is an action to specify the action where data is required. So if a users cache does expire what happens? I might have a look into this one. As a minimum I would change the error to something more meaningful. I.e. required data not found. I need to think about this one as it may be a case of sending the user to the what you need page rather than throwing an exception. Let me look into this.

Reply:
Just update the error to "Required data not found" for now and I will do some testing on this.